pkg/clustermesh/cell.go

```diff
@@ -39,8 +39,14 @@ var Cell = cell.Module(
 	metrics.Metric(NewMetrics),
 	metrics.Metric(common.MetricsProvider(subsystem)),
 
-	cell.Invoke(func(info types.ClusterInfo, dcfg *option.DaemonConfig, cnimgr cni.CNIConfigManager) error {
-		return info.ValidateBuggyClusterID(dcfg.IPAM, cnimgr.GetChainingMode())
+	cell.Config(types.DefaultQuirks),
+	cell.Invoke(func(info types.ClusterInfo, dcfg *option.DaemonConfig, cnimgr cni.CNIConfigManager, cfg Configuration, quirks types.QuirksConfig) error {
```

Comment:
```suggestion
	cell.Invoke(func(logger logrus.FieldLogger, info types.ClusterInfo, dcfg *option.DaemonConfig, cnimgr cni.CNIConfigManager, quirks types.QuirksConfig) error {
```

Nit. I'd suggest depending explicitly on the logger, rather than on the broader cfg object.

Reply:
How do we make sure the Logger is the right one, for example including the right subsystem etc? Do we need to declare a specific clustermesh logger type and provide it from the same Cell?

Reply:
That's automatically taken care of by hive, which always provides a logger (either logrus or slog) scoped for the module that requested it. The logger part of the `Configuration` structure is also already retrieved that way, without any further customization.

Reply:
It looks like Hive/Cell magically populates this. Incremental patch, will push soon:

```
diff --git a/pkg/clustermesh/cell.go b/pkg/clustermesh/cell.go
index 9cc5ba6591f2..d3f2c7cd2ffe 100644
--- a/pkg/clustermesh/cell.go
+++ b/pkg/clustermesh/cell.go
@@ -5,6 +5,7 @@ package clustermesh
 
 import (
        "github.com/cilium/hive/cell"
+       "github.com/sirupsen/logrus"
 
        "github.com/cilium/cilium/daemon/cmd/cni"
        "github.com/cilium/cilium/pkg/clustermesh/common"
@@ -40,10 +41,10 @@ var Cell = cell.Module(
        metrics.Metric(common.MetricsProvider(subsystem)),
 
        cell.Config(types.DefaultQuirks),
-       cell.Invoke(func(info types.ClusterInfo, dcfg *option.DaemonConfig, cnimgr cni.CNIConfigManager, cfg Configuration, quirks types.QuirksConfig) error {
+       cell.Invoke(func(info types.ClusterInfo, dcfg *option.DaemonConfig, cnimgr cni.CNIConfigManager, log logrus.FieldLogger, quirks types.QuirksConfig) error {
                err := info.ValidateBuggyClusterID(dcfg.IPAM, cnimgr.GetChainingMode())
                if err != nil && quirks.AllowUnsafePolicySKBUsage {
-                       cfg.Logger.WithError(err).Error("Detected clustermesh ID configuration that may cause connection impact")
+                       log.WithError(err).Error("Detected clustermesh ID configuration that may cause connection impact")
                        return nil
                }
                return err
```